wled00/bus_manager.cpp

```diff
@@ -605,6 +605,198 @@ void BusNetwork::cleanup() {
+      mxconfig.mx_height = 32;
+      break;
+    case 102:
+      mxconfig.mx_width = 64;
+      mxconfig.mx_height = 32;
+      break;
+    case 103:
+      mxconfig.mx_width = 64;
+      mxconfig.mx_height = 64;
+      break;
+  }
+
+  mxconfig.chain_length = max((u_int8_t) 1, min(bc.pins[0], (u_int8_t) 4)); // prevent bad data preventing boot due to low memory
+
+  if(mxconfig.mx_width >= 64 && (bc.pins[0] > 1)) {
+    DEBUG_PRINTF("WARNING, only single panel can be used of 64 pixel boards due to memory")
+    mxconfig.chain_length = 1;
+  }
+
+  // mxconfig.driver   = HUB75_I2S_CFG::SHIFTREG;
+
+#if defined(ARDUINO_ADAFRUIT_MATRIXPORTAL_ESP32S3) // MatrixPortal ESP32-S3
+
+  // https://www.adafruit.com/product/5778
+
+  DEBUG_PRINTF("MatrixPanel_I2S_DMA - Matrix Portal S3 config");
+
+  mxconfig.gpio.r1 = 42;
+  mxconfig.gpio.g1 = 41;
+  mxconfig.gpio.b1 = 40;
+  mxconfig.gpio.r2 = 38;
+  mxconfig.gpio.g2 = 39;
+  mxconfig.gpio.b2 = 37; 
+
+  mxconfig.gpio.lat = 47;
+  mxconfig.gpio.oe  = 14;
+  mxconfig.gpio.clk = 2;
+
+  mxconfig.gpio.a = 45;
+  mxconfig.gpio.b = 36;
+  mxconfig.gpio.c = 48;
+  mxconfig.gpio.d = 35;
+  mxconfig.gpio.e = 21;
+
+#elif defined(ESP32_FORUM_PINOUT) // Common format for boards designed for SmartMatrix
+
+  DEBUG_PRINTLN("MatrixPanel_I2S_DMA - ESP32_FORUM_PINOUT");
+
+/*
+    ESP32 with SmartMatrix's default pinout - ESP32_FORUM_PINOUT
+    
+    https://github.com/pixelmatix/SmartMatrix/blob/teensylc/src/MatrixHardware_ESP32_V0.h
+
+    Can use a board like https://github.com/rorosaurus/esp32-hub75-driver
+*/
+
+  mxconfig.gpio.r1 = 2;
+  mxconfig.gpio.g1 = 15;
+  mxconfig.gpio.b1 = 4;
+  mxconfig.gpio.r2 = 16;
+  mxconfig.gpio.g2 = 27;
+  mxconfig.gpio.b2 = 17; 
+
+  mxconfig.gpio.lat = 26;
+  mxconfig.gpio.oe  = 25;
+  mxconfig.gpio.clk = 22;
+
+  mxconfig.gpio.a = 5;
+  mxconfig.gpio.b = 18;
+  mxconfig.gpio.c = 19;
+  mxconfig.gpio.d = 21;
+  mxconfig.gpio.e = 12;
+
+#else
+  DEBUG_PRINTLN("MatrixPanel_I2S_DMA - Default pins");
+  /*
+   https://github.com/mrfaptastic/ESP32-HUB75-MatrixPanel-DMA?tab=readme-ov-file
+
+   Boards
+
+   https://esp32trinity.com/
+   https://www.electrodragon.com/product/rgb-matrix-panel-drive-interface-board-for-esp32-dma/
+   
+  */
+  mxconfig.gpio.r1 = 25;
+  mxconfig.gpio.g1 = 26;
+  mxconfig.gpio.b1 = 27;
+  mxconfig.gpio.r2 = 14;
+  mxconfig.gpio.g2 = 12;
+  mxconfig.gpio.b2 = 13;
+
+  mxconfig.gpio.lat = 4;
+  mxconfig.gpio.oe  = 15;
+  mxconfig.gpio.clk = 16;
+
+  mxconfig.gpio.a = 23;
+  mxconfig.gpio.b = 19;
+  mxconfig.gpio.c = 5;
+  mxconfig.gpio.d = 17;
+  mxconfig.gpio.e = 18;
+
+#endif
+
+
+  DEBUG_PRINTF("MatrixPanel_I2S_DMA config - %ux%u length: %u\n", mxconfig.mx_width, mxconfig.mx_height, mxconfig.chain_length);
+
+  // OK, now we can create our matrix object
+  display = new MatrixPanel_I2S_DMA(mxconfig);
+
+  this->_len = (display->width() * display->height());
+
+  pinManager.allocatePin(mxconfig.gpio.r1, true, PinOwner::HUB75);
```

Comment:
There is also an `allocateMultiplePins()` method available.

Reply:
It's a struct for the gpio pins, not array so can't really swap over easily